lib/types/subnet_info.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Types.SubnetInfo do
+  @moduledoc """
+  Struct to hold subnet attestations for easier db storing:
+  - data: An attestation data.
+  - attestations: List of all the collected Attestations.
+  """
+
+  defstruct [:data, :attestations]
+
+  @type t :: %__MODULE__{
+          data: Types.AttestationData.t(),
+          attestations: list(Types.Attestation.t())
+        }
+
+  @spec new_subnet_with_attestation(Types.Attestation.t()) :: Types.SubnetInfo.t()
+  def new_subnet_with_attestation(%Types.Attestation{data: data} = attestation),
+    do: %__MODULE__{data: data, attestations: [attestation]}
+
+  @spec aggregate_attestation(t(), Types.Attestation.t()) :: t()
+  def aggregate_attestation(subnet_info, attestation) do
+    if subnet_info.data == attestation.data do
+      %__MODULE__{subnet_info | attestations: [attestation | subnet_info.attestations]}
```

Comment:
Maybe this is for a different PR (let's start an issue if that's the case), but if we only save attestations with the same data, do we really need to save all attestation structs in a list? We could actually store only the signature and aggregation bits for each one. Either that or we aggregate in-place and hold only one attestation instead of a list.

Reply:
Created #1142!